editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/ArgumentStyleChildTableControl.java

```diff
@@ -0,0 +1,416 @@
+			this.editButton.setEnabled(enable);
+			this.removeButton.setEnabled(enable);
+		}
+	}
+
+	/**
+	 * @return the current selection from the table
+	 */
+	public IStructuredSelection getStructuredSelection() {
+		if (propertyTreeTable != null && !propertyTreeTable.getSelection().isEmpty()) {
+			return (IStructuredSelection) propertyTreeTable.getSelection();
+		}
+		return null;
+	}
+
+	/**
+	 * @return warning string
+	 */
+	public String getWarning() {
+		return this.warningMsg;
+	}
+
+	protected void removeArgumentFromList() {
+		if (!getStructuredSelection().isEmpty()) {
+			AbstractCamelModelElement selectedProperty = (AbstractCamelModelElement) getStructuredSelection()
+					.getFirstElement();
+			argumentList.remove(selectedProperty);
+		}
+	}
+
+	protected void addArgumentTypeToList() {
+		final ArgumentInputDialog dialog = new ArgumentInputDialog(Display.getCurrent().getActiveShell());
+		int rtnValue = dialog.open();
+		if (rtnValue == PropertyInputDialog.OK) {
+			final String type = dialog.getArgumentType();
+			final String value = dialog.getArgumentValue();
+			addBeanArgument(type, value);
+		}
+	}
+
+	protected void editArgumentType() {
+
+		if (!getStructuredSelection().isEmpty()) {
+			AbstractCamelModelElement selectedProperty = (AbstractCamelModelElement) getStructuredSelection()
+					.getFirstElement();
+
+			final ArgumentInputDialog dialog = new ArgumentInputDialog(Display.getCurrent().getActiveShell());
+			final Element xmlElement = (Element) selectedProperty.getXmlNode();
+			if (xmlElement.getAttribute(TYPE_COLUMN) != null) {
+				dialog.setArgumentType(xmlElement.getAttribute(TYPE_COLUMN));
+			}
+			if (xmlElement.getAttribute(VALUE_COLUMN) != null) {
+				dialog.setArgumentValue(xmlElement.getAttribute(VALUE_COLUMN));
+			}
+			int rtnValue = dialog.open();
+			if (rtnValue == PropertyInputDialog.OK) {
+				final String type = dialog.getArgumentType();
+				final String value = dialog.getArgumentValue();
+				if (!Strings.isEmpty(type)) {
+					xmlElement.setAttribute(TYPE_COLUMN, type);
+				} else {
+					xmlElement.removeAttribute(TYPE_COLUMN);
+				}
+				if (!Strings.isEmpty(value)) {
+					xmlElement.setAttribute(VALUE_COLUMN, value);
+				}
+			}
+		}
+	}
+
+	public void setInput(Object input) {
+		if (input instanceof AbstractCamelModelElement) {
+			inputElement = (AbstractCamelModelElement) input;
+		}
+	}
+
+	private void addBeanArgument(String type, String value) {
+		String prefixNS = null;
+		boolean isBlueprint = false;
+		AbstractCamelModelElement newRoot = inputElement;
+		if (inputElement instanceof CamelFile && !inputElement.getChildElements().isEmpty()) {
+			newRoot = inputElement.getChildElements().get(0);
+			prefixNS = newRoot.getRouteContainer().getXmlNode().getPrefix();
+			isBlueprint = ((CamelFile) inputElement).isBlueprint();
+		} else if (inputElement.getRouteContainer() != null) {
+			prefixNS = inputElement.getRouteContainer().getXmlNode().getPrefix();
```

Comment:
Good catch. I can add a call to isBlueprint = inputElement.getCamelFile().isBlueprint(); to set that more consistently.